src/lib/ThemeContext.tsx

```diff
@@ -91,13 +96,31 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({
   children,
 }) => {
   const colorSchemeFromRN = useColorScheme()
+  const [nativeColorScheme, setNativeColorScheme] =
+    React.useState<ColorSchemeName>(colorSchemeFromRN)
+  const nativeColorSchemeChangeTimeout = React.useRef<NodeJS.Timeout | null>(
+    null,
+  )
 
-  // if theme is 'system', use the device's configured color scheme
-  let colorScheme = theme === 'system' ? colorSchemeFromRN : theme
+  React.useEffect(() => {
+    if (nativeColorSchemeChangeTimeout.current)
+      clearTimeout(nativeColorSchemeChangeTimeout.current)
+    nativeColorSchemeChangeTimeout.current = setTimeout(
+      () => setNativeColorScheme(colorSchemeFromRN),
+      1000,
+    )
+  }, [colorSchemeFromRN])
 
   const value = useMemo(
```

Comment:
Why is this `useMemo` at all? It doesn't do any expensive calculation. Let's remove it and make this a plain statement.

Reply:
Oh no reason, was just there before. I'll remove 👍